test/specs/utils/deepMerge.spec.js

```diff
@@ -53,14 +53,26 @@ describe('utils::deepMerge', function () {
     expect(d.foo).not.toBe(a.foo);
   });
 
+  it('Should not change the data type', function() {
+    var a = {foo: {bar:123}, hnd: [444, 555, 666]};
+    var b = {foo: {baz: 456}, hnd: [222]};
+    var d = deepMerge({}, a, b);
+    expect(d).toEqual({
+      foo: {bar: 123, baz: 456},
+      hnd: [222, 555, 666]
```

Comment:
Personally, I prefer simply replacement for arrays, instead of merging. Because merging the same index values looks less meaningful.

Reply:
Merge arrays to maintain the same behavior as the original 'deepMerge' function.

Reply:
Yes, but the original behavior is in v0.18, which is `merge` in lib/utils.js.